src/Aspire.Hosting.Azure.WebPubSub/AzureWebPubSubExtensions.cs

```diff
@@ -166,6 +167,31 @@ public static IResourceBuilder<AzureWebPubSubHubResource> AddHub(this IResourceB
         return hubBuilder;
     }
 
+    /// <summary>
+    /// Adds an Azure Web Pub Sub hub resource to the application model.
+    /// </summary>
+    /// <param name="builder">The Azure WebPubSub resource builder.</param>
+    /// <param name="name">The name of the Azure WebPubSub Hub resource.</param>
+    /// <param name="hubName">The name of the Azure WebPubSub Hub. If not provided, this defaults to the same value as <paramref name="name"/>.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureWebPubSubHubResource> AddHub(this IResourceBuilder<AzureWebPubSubResource> builder, [ResourceName] string name, string? hubName = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        // Use the resource name as the hub name if it's not provided
+        hubName ??= name;
+
+        AzureWebPubSubHubResource? hubResource;
+        if (!builder.Resource.Hubs.TryGetValue(hubName, out hubResource))
+        {
+            hubResource = new AzureWebPubSubHubResource(hubName, builder.Resource);
```

Comment:
You actually want to split the properties out on the child resource, like we do for EventHubs:

https://github.com/dotnet/aspire/blob/5fd2b10c5fc644ae3aac08a0ca895d7fd72af86c/src/Aspire.Hosting.Azure.EventHubs/AzureEventHubResource.cs#L19-L25

The `name` is the resource name (and can only have letters, numbers, and hyphens), and can't be duplicated in the model. You can only have 1 resource named "foo", even if they have different parents.